app/models/user.rb

```diff
@@ -45,6 +45,9 @@ class User < ApplicationRecord
 
 	validates :name, :presence => true
 
+	validate :validate_pin
```

Comment:
you could more succintly do validate :validate_pin, unless: pin.empty?

Reply:
yes, this will now reduce the check in validator. Thanks !!